File: app/src/main/java/com/jdappel/beerinvestigator/rest/BreweryDBResponse.java
package com.jdappel.beerinvestigator.rest;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.List;

/**
 * Created by jappel on 2/20/2016.
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public class BreweryDBResponse<T> {

    private final String status;
    private final List<T> data;

    @JsonCreator
    public BreweryDBResponse(@JsonProperty("success") final String status, @JsonProperty("data") final List<T> data) {
        this.status = status;
        this.data = data;
    }

    public String getStatus() {
        return status;
    }

    public List<T> getData() {
        return data;
    }

    @Override
    public String toString() {
        return "BreweryDBResponse{" +
                "status='" + status + '\'' +
                ", data=" + data +
                '}';
    }
}

File: app/src/main/java/com/jdappel/beerinvestigator/data/rest/BreweryDBResponse.kt
package com.jdappel.beerinvestigator.data.rest

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty

/**
 * Generic Type for returning JSON data from BreweryDB
 */
@JsonIgnoreProperties(ignoreUnknown = true)
class BreweryDBResponse<T> @JsonCreator constructor(
    @param:JsonProperty("success") val status: String? = null, @param:JsonProperty(
        "data"
    ) val data: List<T>?
) {
    override fun toString(): String {
        return "BreweryDBResponse{" +
                "status='" + status + '\'' +
                ", data=" + data +
                '}'
    }
}